### DISCUSSION TITLE
Can't change the returned value of a many-to-many-connection

### DISCUSSION THREAD

**Discussion by deraleks1** [directus/directus#7214] [Archive]
### Preflight Checklist

- [X] I have completed all [Troubleshooting Steps](https://docs.directus.io/getting-started/support/#troubleshooting-steps).
- [X] I'm on [the latest version of Directus](https://github.com/directus/directus/releases).
- [X] There's [no other issue](https://github.com/directus/directus/issues) that already describes my problem.

### Describe the Bug

I have created a many-to-many-connection between two collections:
- A collection listing various exercises
- A collection listing affected muscle groups
An exercise item can contain 1-n muscle groups.
The problem is that when you query that exercise-item then the primarry_key of the conjoined database is returrned rather than the primary_key of the muscle-group-collection (which is what we want). Is such a concept possible in directus?

### To Reproduce

- create 2 collection (A & B)
- create dummy data in collection B (just a name field for example)
- create a many-to-many-connection in collection A
- create an item in collection A connecting items from collection B
- inspect the raw-values of the connection A (it should be the primary_key of the conjoined collection)

### What version of Directus are you using?

cloud

### What version of Node.js are you using?

cloud

### What database are you using?

cloud

### What browser are you using?

Chrome

### What operating system are you using?

Winddows

### How are you deploying Directus?

cloud

**Comment by rijkvanzanten** [comment]
Many-to-many fields are simply a one-to-many to a junction table, followed by a many-to-one to the related table. You can use [the `fields` parameter](https://docs.directus.io/reference/api/query/#fields) to specify what fields, and in what depth, are returned 👍🏻

**Comment by deraleks1** [comment]
I have made some screenshots to visual my issue:
I have created an API call to look what is returrned:

![](https://i.imgur.com/KC222B6.jpeg)

The first element ID describes this:


![](https://i.imgur.com/vkLFHhU.jpeg)

In this case I want to have "1" returned. Here is my config of the connection ("Name" in interface and display is the focus-group-name value):

![](https://i.imgur.com/JTzyAZo.jpeg)

![](https://i.imgur.com/Iul4e7S.jpeg)

![](https://i.imgur.com/YzIvRSs.jpeg)

![](https://i.imgur.com/gJwk3Hn.jpeg)

![](https://i.imgur.com/9OpTUVH.jpeg)

![](https://i.imgur.com/Mkyo2wF.jpeg)

**Reply by rijkvanzanten** [reply]
Yeah, I'm aware! That's currently not possible. Like I mentioned above, a many-to-many field is nothing more than a one-to-many followed by a many-to-one, so you'll fetch the junction table first, and then the related items. You can use the `fields` parameter to specific what fields are returned in the response. Try using `fields=qwerty.focus_group_id`

**Reply by deraleks1** [reply]
That fixes it for now. Thank you very much :)
